### PR TITLE
Remove untraced comm event

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
Provide a brief description of what this PR does, and why it's needed.

Paired with @goral09 and @odeac 

We do not need the untraced comm event because if there is a comm event that can happen but is not in the replay logs, then the resulting replay rspace's tuplespace/state hash will not match the hash in the block that contains the replay logs. We need to remove the untraced comm event because of the following case:

            for (@winner <- @"ch") {
              @"return"!(winner)
            } | @"ch"!(1) | @"ch"!(2)

The replay logs say that `for (@winner <- @"ch")` matches with `@"ch"!(1)`. Assume the replay rspace then gets called in the order of 1) `for (@winner <- @"ch")`, 2) `@"ch"!(2)`, and 3) `@"ch"!(1)` . Previously it would complain that `@"ch"!(2)` can match with `for (@winner <- @"ch")` and thus it would generate an untraced comm event. However, `for (@winner <- @"ch")` was just waiting for the `@"ch"!(1)` that was about to come.

Integration tests pass as below

```

collected 4 items                                                                                                                     

test/network/test_basics.py::test_metrics_api_socket PASSED                                                                     [ 25%]
test/network/test_basics.py::test_node_logs_for_errors PASSED                                                                   [ 50%]
test/network/test_basics.py::test_node_logs_for_RuntimeException PASSED                                                         [ 75%]
test/network/test_casper_propose_and_deploy.py::test_casper_propose_and_deploy PASSED                                           [100%]

===================================================== 4 passed in 196.43 seconds ======================================================
```

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
Add link to corresponding JIRA issue.

https://rchain.atlassian.net/browse/RHOL-694

### Complete this checklist before you submit the PR
- [X] This PR contains no more than 200 lines of code, excluding test code.
- [X] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [X] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

**Comment by odeac** [comment]
Looking at your changes I don't understand why the crash happened when the list length contained 4,7,11... items. Can you explain that?

File: rspace/src/test/scala/coop/rchain/rspace/ReplayRSpaceTests.scala [hunk 1]
```diff
@@ -57,20 +57,6 @@ trait ReplayRSpaceTests extends ReplayRSpaceTestsBase[String, Pattern, String, S
       space.store.isEmpty shouldBe true
   }
 
-  "Creating a COMM Event that is not contained in the trace log" should "throw a ReplayException" in
```
Comment (pyrocto) [inline rspace/src/test/scala/coop/rchain/rspace/ReplayRSpaceTests.scala]: This test should still pass.

Comment (lukasz-golebiewski) [reply rspace/src/test/scala/coop/rchain/rspace/ReplayRSpaceTests.scala]: I agree. Why is it being removed?

**Comment by birchmd** [comment]
I think what these changes show is that the logic for detecting an untraced comm event is not right yet. I think rather than removing the logic entirely we should change it so that the checks are deferred to the end and we only need to check
1) The final state is quiescent (if not then any comm events are flagged as untraced)
2) All comm events in the log have already happened.

File: casper/src/test/scala/coop/rchain/casper/util/rholang/InterpreterUtilTest.scala [hunk 2]
```diff
@@ -505,4 +505,47 @@ class InterpreterUtilTest
 
     tsHash should be(Some(computedTsHash))
   }
+
+  "validateBlockCheckpoint" should "pass tests involving pick" in {
```
Comment (lukasz-golebiewski) [inline casper/src/test/scala/coop/rchain/casper/util/rholang/InterpreterUtilTest.scala]: I ran this test against latest `dev` and it passes. Can you help me understand why the changes in ReplayRSpace are needed?

Comment (lukasz-golebiewski) [reply casper/src/test/scala/coop/rchain/casper/util/rholang/InterpreterUtilTest.scala]: OK, I see now that it fails randomly

**Comment by KentShikama** [comment]
@birchmd checking at the end is the same thing as just checking if the state hashes match. If there were any untraced comm events, then they wouldn't match.

File: rspace/src/test/scala/coop/rchain/rspace/ReplayRSpaceTests.scala (hunk 1 above)
Comment (KentShikama) [reply rspace/src/test/scala/coop/rchain/rspace/ReplayRSpaceTests.scala]: No it just doesn't reduce and thus at the end the state hashes will mismatch.

Comment (KentShikama) [reply rspace/src/test/scala/coop/rchain/rspace/ReplayRSpaceTests.scala]: Think about the case when there is a produce and consume that could reduce but they were originally part of different comm events.

**Comment by lukasz-golebiewski** [comment]
@KentShikama This means that you don't need ReplayRSpace at all, you could use normal RSpace on a different branch

File: rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala [hunk 3]
```diff
@@ -161,35 +144,25 @@ class ReplayRSpace[C, P, A, R, K](store: IStore[C, P, A, K], branch: Branch)(
...
-                    store.getData(txn, Seq(c)).zipWithIndex.filter {
-                      case (Datum(_, _, source), _) => comm.produces.contains(source)
-                    }
-                  case None =>
-                    Random.shuffle(store.getData(txn, Seq(c)).zipWithIndex)
+                val as = store.getData(txn, Seq(c)).zipWithIndex.filter {
+                  case (Datum(_, _, source), _) => comm.produces.contains(source)
                 }
-                c -> { if (c == channel) (Datum(data, persist, produceRef), -1) +: as else as }
+                c -> { if (c == channel) Seq((Datum(data, persist, produceRef), -1)) else as }
```
Comment (KentShikama) [inline rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala]: To anyone reviewing: extra care on this change

Comment (KentShikama) [reply rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala]: I'm assuming that `as` is never going to contain any elements if `c == channel`

**Comment by KentShikama** [comment]
@odeac We still don't really know why it only failed on 4, 7, 11, etc.

**Comment by KentShikama** [comment]
@odeac I would say at this point it is just noise / the way memory is jiggered, allowing the produce/consume to enter the replay rspace in a way that raised the untraced comm event.
